### ISSUE TITLE
Add optional color to debug groups/markers

### ISSUE THREAD

**Issue by JMS55** [gfx-rs/wgpu#6313]
**Is your feature request related to a problem? Please describe.**
A program with thousands of API calls gets hard to debug in tools like NSight, RenderDoc, or PIX.

**Describe the solution you'd like**
Allow optionally specifying a color for debug group/marker functions to make visually finding the passes your interested in easier.

**Comment by cwfitzgerald** [comment]
Amusingly, this used to be in the api way long ago but was removed. I think we should just ignore any color values unsupported by the underlying api.

**Comment by JMS55** [comment]
Agreed, if the underlying API does not support it, just skip the coloring.
